inc/front.php

```diff
@@ -191,6 +196,11 @@ protected static function setup_links($post)
             return;
         }
 
+        if ( isset( self::$no_links_cache[$post->ID] ) && true === self::$no_links_cache[$post->ID] ) {
+            self::$links = array();
+            return;
+        }
```

Comment:
I don't think this is the right approach for caching, but I do agree it's good to do caching.

Use the `wp_cache_*` functions instead so folks using object caches can get that benefit and everyone else will get the equivalent of of what you have here.

```php
if (true === wp_cache_get("nolinks_{$post->ID}", 'seoal')) {
  self::$links = array();
  return;
}

// ...

if (empty($links)) {
   wp_cache_set("nolinks_{$post->ID}", true, 'seoal');
}
```

This may not be necessary, though. Doesn't `get_post` do some of its own object caching?

Reply:
Some other things to think about:

1. Should we cache the final links completely? There's some processing down below for blacklists that may be beneifitial for caching.
2. This cache check as it is now means that the `seoal_links` filter doesn't run when there are no links (it did previously).